cql3/statements/alter_type_statement.cc

```diff
@@ -87,6 +87,21 @@ void alter_type_statement::do_announce_migration(database& db, ::keyspace& ks, b
         throw exceptions::invalid_request_exception(format("No user type named {} exists.", _name.to_string()));
     }
 
+    for (auto&& schema : ks.metadata()->cf_meta_data() | boost::adaptors::map_values) {
+        for (auto&& column : schema->partition_key_columns()) {
+            if (column.type->references_user_type(_name.get_keyspace(), _name.get_user_type_name())) {
+                throw exceptions::invalid_request_exception(format("Cannot add new field to type {} because it is used in the partition key of table {}",
+                    _name.to_string(), schema->cf_name()));
+            }
+        }
+        for (auto&& column : schema->clustering_key_columns()) {
+            if (column.type->references_user_type(_name.get_keyspace(), _name.get_user_type_name())) {
+                throw exceptions::invalid_request_exception(format("Cannot add new field to type {} because it is used in the clustering key of table {}",
+                    _name.to_string(), schema->cf_name()));
+            }
+        }
```

Comment:
Maybe extract a lambda to avoid code duplication?
```
auto check_columns = [&] (auto& columns) {
        for (auto&& column : schema->partition_key_columns()) {
            if (column.type->references_user_type(_name.get_keyspace(), _name.get_user_type_name())) {
                throw exceptions::invalid_request_exception(format("Cannot add new field to type {} because it is used in the primary key of table {}",
                    _name.to_string(), schema->cf_name()));
            }
        }
}
```

Reply:
Also, in the error message I would include the keyspace of the table and the name of the column that uses the UDT.